tests/test_download_firmware.py

```
@@ -3832,3 +3948,758 @@ def test_download_repo_prerelease_firmware_tracks_only_downloaded_when_failures(
+        old_prerelease = prerelease_dir / "firmware-2.7.20.aaaaaa"
+        old_prerelease.mkdir()
+        latest_link = prerelease_dir / LATEST_POINTER_NAME
+        latest_link.symlink_to(old_prerelease)
+
+        with patch.object(
+            downloader,
+            "_get_release_storage_tag",
+            return_value="v2.7.22",
+            create=True,
+        ):
+            with patch.object(log_utils, "logger") as mock_logger:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)